dsn.go

```diff
@@ -117,7 +117,7 @@ func DSN(cfg *Config) (dsn string, err error) {
 			params.Add(k, *v)
 		}
 	}
-	dsn = fmt.Sprintf("%v:%v@%v:%v", cfg.User, cfg.Password, cfg.Host, cfg.Port)
+	dsn = fmt.Sprintf("%v:%v@%v:%v", url.QueryEscape(cfg.User), url.QueryEscape(cfg.Password), cfg.Host, cfg.Port)
```

Comment:
@tjj5036 this is the fix.
Username and password may include `:` and `@`.

Reply:
That makes sense to me... provided this doesn't break DSN parsing (which it looks like it doesn't since all tests pass) then I'd say this is great!